src/etc/inc/system.inc

```diff
@@ -1780,10 +1786,28 @@ function system_ntp_configure() {
 	}
 	@unlink("{$g['varrun_path']}/ntpd.pid");
 
+	/* set NTP server authentication key */
+	if ($config['ntpd']['serverauth'] == 'yes') {
+		$ntpkeyscfg = "1 " . strtoupper($config['ntpd']['serverauthalgo']) . " " . base64_decode($config['ntpd']['serverauthkey']) . "\n";
+		if (!@file_put_contents("{$g['varetc_path']}/ntp.keys", $ntpkeyscfg)) {
+			log_error(sprintf(gettext("Could not open %s/ntp.keys for writing"), $g['varetc_path']));
+			return;
+		}
+	}
```

Comment:
If this auth is disabled, the code should check to see if the `ntp.keys` file is present and remove it.

That or unconditionally do an `unlink_if_exists()` on it before the `if` statement.

Reply:
Updated